src/types/index.ts

```diff
@@ -57,31 +57,31 @@ export type Answer = {
 };
 
 export type Step = {
-  type: 'exercise';
-  task: Task;
+  type?: 'exercise';
+  task?: Task;
   uid: ExerciseUid;
   id: number;
   available_points: AvailablePoints;
-  preview: string;
+  preview?: string;
   is_completed: boolean;
   answer_id?: ID;
   answer_id_order: ID[];
   free_response: '';
   feedback_html: '';
```

Comment:
I think we need all the question specific stuff to be optional in this type

Reply:
Or maybe we can just change the props here https://github.com/openstax/assessment-components/pull/21/files#diff-04fbf9f4c5012dd634f925890f1e6be5f9442a5bf710e86e677bfcccc3e4dfc8R45 to apply some different stuff from the exercise data